Mantle/MTLModel+Difference.m

```diff
@@ -10,11 +10,29 @@ @implementation MTLModel (Difference)
 - (NSArray<NSString *> *)differenceFrom:(id)object {
   MTLModel *actual = (MTLModel *)object;
 
-  NSAssert2([object isKindOfClass:self.class], @"Expected object of type %@, got : %@",
-            self.class, [object class]);
+  if (!actual) {
+    return @[
+      [NSString stringWithFormat:@"Expected: object of type %@, got: nil", NSStringFromClass(self.class)]
+    ];
+  }
 
   NSMutableArray<NSString *> *differences = [NSMutableArray array];
+  if (![actual isKindOfClass:self.class]) {
```

Comment:
You probably want to know if it's exactly the same class, not just inherits from it (Use `isEqual:`). Add a test for this.

Reply:
Originally, I did intended to support difference between related classes, but I'm getting more and more discouraged as the time passes. So I'll use class equality as you suggested and we'll wait for the first real need for something more elaborate.